extensions/amp-form/0.1/amp-form.js

```diff
@@ -255,6 +272,30 @@ export class AmpForm {
   }
 
   /**
+   * Checks whether the submissions are going to go through to the canonical origin
+   * or not.
+   * @private
+   */
+  isSubmittingToCanonical_() {
+    if (this.isCanonicalAction_ !== undefined) {
+      return this.isCanonicalAction_;
+    }
+
+    const docInfo = documentInfoForDoc(this.form_);
+    const canonicalOrigin = parseUrl(docInfo.canonicalUrl).origin;
```

Comment:
Using `getSourceOrigin` might give false positives here - although that case is validated but I'd rather have a redundant use-case here. In general we don't allow `cdn.ampproject.org` urls to be a form action, so in this case we'd get the source origin although the URL would submit to `cdn.ampproject.org` origin.